includes/class-mailer.php

```diff
@@ -111,26 +115,73 @@ public static function new_follower( $notification ) {
+		}
+
+		/* translators: 1: Blog name, 2: Follower name */
+		$subject = \sprintf( \esc_html__( '[%1$s] Follower: %2$s', 'activitypub' ), \esc_html( get_option( 'blogname' ) ), \esc_html( $actor['name'] ) );
+		/* translators: 1: Blog name, 2: Follower name */
+		$message = \sprintf( \esc_html__( 'New Follower: %2$s.', 'activitypub' ), \esc_html( get_option( 'blogname' ) ), \esc_html( $actor['name'] ) ) . "\r\n\r\n";
+		/* translators: Follower URL */
+		$message .= \sprintf( \esc_html__( 'URL: %s', 'activitypub' ), \esc_url( $actor['url'] ) ) . "\r\n\r\n";
+		$message .= \esc_html__( 'You can see all followers here:', 'activitypub' ) . "\r\n";
+		$message .= \esc_url( \admin_url( $admin_url ) ) . "\r\n\r\n";
+
+		\wp_mail( $email, $subject, $message );
+	}
+
+	/**
+	 * Send a direct message.
+	 *
+	 * @param array $activity The activity object.
+	 * @param int   $user_id  The id of the local blog-user.
+	 */
+	public static function direct_message( $activity, $user_id ) {
+		// Check if Activity is public or not.
+		if (
+			is_activity_public( $activity ) &&
+			is_activity_reply( $activity )
+		) {
+			return;
+		}
+
+		$actor = get_remote_metadata_by_actor( $activity['actor'] );
+
+		if ( ! $actor || \is_wp_error( $actor ) || empty( $activity['object']['content'] ) ) {
+			return;
+		}
+
+		$email = \get_option( 'admin_email' );
+
+		if ( (int) $user_id > Actors::BLOG_USER_ID ) {
+			$user = \get_user_by( 'id', $user_id );
+
+			if ( ! $user ) {
+				return;
+			}
+
 			$email = $user->user_email;
 		}
 
-		/* translators: %1$s: Blog name, %2$s: Follower name */
-		$subject = \sprintf( \__( '[%1$s] Follower: %2$s', 'activitypub' ), get_option( 'blogname' ), $actor['name'] );
-		/* translators: %1$s: Blog name, %2$s: Follower name */
-		$message = \sprintf( \__( 'New follower: %2$s', 'activitypub' ), get_option( 'blogname' ), $actor['name'] ) . "\r\n";
-		/* translators: %s: Follower URL */
-		$message .= \sprintf( \__( 'URL: %s', 'activitypub' ), $actor['url'] ) . "\r\n\r\n";
-		$message .= \sprintf( \__( 'You can see all followers here:', 'activitypub' ) ) . "\r\n";
-		$message .= \esc_url( \admin_url( '/users.php?page=activitypub-followers-list' ) ) . "\r\n\r\n";
+		/* translators: 1: Blog name, 2 Actor name */
+		$subject = \sprintf( \esc_html__( '[%1$s] Direct Message from: %2$s', 'activitypub' ), \esc_html( get_option( 'blogname' ) ), \esc_html( $actor['name'] ) );
+		/* translators: 1: Blog name, 2: Actor name */
+		$message = \sprintf( \esc_html__( 'New Direct Message: %2$s', 'activitypub' ), \esc_html( get_option( 'blogname' ) ), \wp_strip_all_tags( $activity['object']['content'] ) ) . "\r\n\r\n";
+		/* translators: Actor name */
+		$message .= \sprintf( \esc_html__( 'From: %s', 'activitypub' ), \esc_html( $actor['name'] ) ) . "\r\n";
+		/* translators: Actor URL */
+		$message .= \sprintf( \esc_html__( 'URL: %s', 'activitypub' ), \esc_url( $actor['url'] ) ) . "\r\n\r\n";
 
 		\wp_mail( $email, $subject, $message );
```

Comment:
Should we support HTML mails for direct messages?

```suggestion
		$message = \sprintf( \esc_html__( 'New Direct Message: %2$s', 'activitypub' ), \esc_html( get_option( 'blogname' ) ), \wp_kses( $activity['object']['content'] ) ) . "<br/><br/>";
		/* translators: Actor name */
		$message .= \sprintf( \esc_html__( 'From: %s', 'activitypub' ), \esc_html( $actor['name'] ) ) . "<br/>";
		/* translators: Actor URL */
		$message .= \sprintf( \esc_html__( 'URL: %s', 'activitypub' ), \esc_url( $actor['url'] ) ) . "<br/><br/>";

		\wp_mail( $email, $subject, $message, array( 'Content-Type: text/html' ) );
```

Reply:
this is not working code, just to show how it might be possible!

Reply:
I think it's fine as is for now.